worker/src/RTC/SCTP/chunkParameters/CookiePreservativeChunkParameter.cpp

```diff
@@ -0,0 +1,148 @@
+#define MS_CLASS "RTC::SCTP::CookiePreservativeChunkParameter"
+// #define MS_LOG_DEV_LEVEL 3
+
+#include "RTC/SCTP/chunkParameters/CookiePreservativeChunkParameter.hpp"
+#include "Logger.hpp"
+#include "MediaSoupErrors.hpp"
+
+namespace RTC
+{
+	namespace SCTP
+	{
+		/* Class methods. */
+
+		CookiePreservativeChunkParameter* CookiePreservativeChunkParameter::Parse(
+		  const uint8_t* buffer, size_t bufferLength)
+		{
+			MS_TRACE();
+
+			ChunkParameter::ChunkParameterType parameterType;
+			uint16_t parameterLength;
+			uint8_t padding;
+
+			if (!ChunkParameter::IsChunkParameter(
+			      buffer, bufferLength, parameterType, parameterLength, padding))
+			{
+				return nullptr;
+			}
+
+			if (parameterType != ChunkParameter::ChunkParameterType::COOKIE_PRESERVATIVE)
+			{
+				MS_WARN_DEV("invalid Chunk Parameter type");
+
+				return nullptr;
+			}
+
+			return CookiePreservativeChunkParameter::ParseStrict(
+			  buffer, bufferLength, parameterLength, padding);
+		}
+
+		CookiePreservativeChunkParameter* CookiePreservativeChunkParameter::ParseStrict(
+		  const uint8_t* buffer, size_t bufferLength, uint16_t parameterLength, uint8_t padding)
+		{
+			MS_TRACE();
+
+			if (parameterLength != CookiePreservativeChunkParameter::CookiePreservativeChunkParameterLength)
+			{
+				MS_WARN_TAG(
+				  sctp,
+				  "CookiePreservativeChunkParameter Length field must be %zu",
+				  CookiePreservativeChunkParameter::CookiePreservativeChunkParameterLength);
+
+				return nullptr;
+			}
+
+			auto* parameter =
+			  new CookiePreservativeChunkParameter(const_cast<uint8_t*>(buffer), bufferLength);
+
+			// Mark the Parameter as frozen since we are parsing.
+			parameter->Freeze();
+
+			return parameter;
+		}
+
+		CookiePreservativeChunkParameter* CookiePreservativeChunkParameter::Factory(
+		  uint8_t* buffer, size_t bufferLength)
+		{
+			MS_TRACE();
+
+			if (bufferLength < CookiePreservativeChunkParameter::CookiePreservativeChunkParameterLength)
+			{
+				MS_THROW_TYPE_ERROR("too small buffer");
```

Comment:
@Cosmetic, this message resonated in my head as I would expect 'buffer too small'. As said, this is cosmetic, just a comment. Not asking to change.

Reply:
I've changed it.